pkg/services/annotations/accesscontrol/accesscontrol.go

```diff
@@ -84,10 +84,16 @@ func (authz *AuthService) dashboardsWithVisibleAnnotations(ctx context.Context,
 	}
 
 	filters := []any{
-		permissions.NewAccessControlDashboardPermissionFilter(user, dashboardaccess.PERMISSION_VIEW, filterType, authz.features, recursiveQueriesSupported),
+		permissions.NewAccessControlDashboardPermissionFilter(query.SignedInUser, dashboardaccess.PERMISSION_VIEW, filterType, authz.features, recursiveQueriesSupported),
 		searchstore.OrgFilter{OrgId: orgID},
 	}
 
+	if query.DashboardUID != "" {
```

Comment:
I think we should do something similar if `query.DashboardId` is set instead

Reply:
Makes sense as well. I can add this just in case.